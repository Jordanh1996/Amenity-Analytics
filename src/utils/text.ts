export interface Info {
  companyName: string;
  isTalkingAboutCompany: boolean;
  products: string[];
  keywordToSentences: { [key: string]: string[] };
}

export const hasOccurrence = (text: string, keyword: string): boolean => {
  return text.includes(keyword);
};

export const getSentences = (
  text: string,
  delimiter: RegExp | string = /\.[\n \t]/
): string[] => {
  return text.split(delimiter);
};

export const getSentencesWithOccurence = (
  sentences: string[],
  keyword: string
): string[] => {
  const res = [];

  for (let sentence of sentences) {
    if (hasOccurrence(sentence, keyword)) {
      res.push(sentence);
    }
  }

  return Array.from(new Set(res)); // this is used in order to delete repeating sentences
};

export const getInfo = (
  text: string,
  companyName: string,
  products: string[]
): Info => {
  const isTalkingAboutCompany = hasOccurrence(text, companyName);
  const sentences = getSentences(text);
  const keywordToSentences: { [key: string]: string[] } = {};
  for (let product of products) {
    keywordToSentences[product] = getSentencesWithOccurence(sentences, product);
  }

  return {
    companyName,
    isTalkingAboutCompany,
    products,
    keywordToSentences,
  };
};
