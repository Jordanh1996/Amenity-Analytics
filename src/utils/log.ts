import { Info } from './text';

export const logResult = ({
  companyName,
  isTalkingAboutCompany,
  products,
  keywordToSentences,
}: Info): void => {
  console.log(
    `Is the article talking about the company ${companyName} ? ${isTalkingAboutCompany}`
  );

  for (let product of products) {
    console.log(); // new line
    const productSentences = keywordToSentences[product];
    if (productSentences) {
      console.log(`${product} sentences:`);
      for (let sentence of productSentences) {
        console.log(`  *  ${sentence}.`);
      }
    }
  }
};
