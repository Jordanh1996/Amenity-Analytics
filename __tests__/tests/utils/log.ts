import { logResult } from '../../../src/utils/log';

describe('log', () => {
  describe('logResult', () => {
    let output: string;
    const storeLog = (input: string | void) =>
      (output += (input ? input : '') + '\n');
    beforeEach(() => {
      output = '';
      console.log = jest.fn(storeLog);
    });

    test('Log should match snapshot', () => {
      logResult({
        companyName: 'Microsoft',
        isTalkingAboutCompany: true,
        products: ['product1', 'product2'],
        keywordToSentences: { product1: ['sentence1', 'sentence2'] },
      });
      expect(output).toMatchSnapshot();
    });
  });
});
