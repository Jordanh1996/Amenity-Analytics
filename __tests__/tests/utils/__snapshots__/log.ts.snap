// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`log logResult Log should match snapshot 1`] = `
"Is the article talking about the company Microsoft ? true

product1 sentences:
  *  sentence1.
  *  sentence2.

"
`;
